src/nvim/ops.c

```diff
@@ -787,7 +787,8 @@ yankreg_T *copy_register(int name)
   if (copy->y_size == 0) {
     copy->y_array = NULL;
   } else {
-    copy->y_array = xcalloc(copy->y_size, sizeof(char_u *));
+    assert(copy->y_size >= 0);
```

Comment:
Unless we check every time we read `yankreg_T->y_size` I don't think this really helps with robustness. If we feel the need, It is better to assert when writing y_size (when it is the result of a nontrivial calculation), which you already do in some place.